File: src/main/java/raylras/zen/langserver/StandardIOLauncher.java
package raylras.zen.langserver;

import org.eclipse.lsp4j.jsonrpc.Launcher;
import org.eclipse.lsp4j.launch.LSPLauncher;
import org.eclipse.lsp4j.services.LanguageClient;

public class StandardIOLauncher {

    public static void main(String[] args) {
        start();
    }

    public static void start() {
        try {
            ZenLanguageServer server = new ZenLanguageServer();
            Launcher<LanguageClient> launcher = LSPLauncher.createServerLauncher(server, System.in, System.out);
            LanguageClient client = launcher.getRemoteProxy();
            server.connect(client);
            launcher.startListening().get();
        } catch (Exception e) {
            e.printStackTrace();
        }
    }

}

File: language-server/src/main/kotlin/raylras/intellizen/languageserver/StandardIOLauncher.kt
package raylras.intellizen.languageserver

import org.eclipse.lsp4j.launch.LSPLauncher
import org.eclipse.lsp4j.services.LanguageClient
import java.util.concurrent.ExecutorService
import java.util.concurrent.Executors
import java.util.function.Function

object StandardIOLauncher {
    val POOL: ExecutorService = Executors.newCachedThreadPool()
    var CLIENT: LanguageClient? = null

    @JvmStatic
    fun main(args: Array<String>) {
        start()
    }

    fun start() {
        val server = ZenLanguageServer(ZenLanguageService())
        val launcher = LSPLauncher.createServerLauncher(server, System.`in`, System.out, POOL, Function.identity())
        CLIENT = launcher.remoteProxy
        launcher.startListening().get()
    }

    fun shutdown() {
        POOL.shutdown()
    }
}
